packages/aws-cdk/lib/version.ts

```diff
@@ -23,23 +27,39 @@ function commit(): string {
   return require('../build-info.json').commit;
 }
 
-export class TimestampFile {
+export class VersionCheckTTL {
+  public static timestampFilePath(): string {
+    // Get the home directory from the OS, first. Fallback to $HOME.
+    const homedir = os.userInfo().homedir || os.homedir();
+    if (!homedir || !homedir.trim()) {
+      throw new Error('Cannot determine home directory');
+    }
+    // Using the same path from account-cache.ts
+    return path.join(homedir, '.cdk', 'cache', 'repo-version-ttl');
+  }
+
   private readonly file: string;
 
-  // File modify times are accurate only till the second, hence using seconds as precision
+  // File modify times are accurate only to the second
   private readonly ttlSecs: number;
 
-  constructor(file: string, ttlSecs: number) {
-    this.file = file;
-    this.ttlSecs = ttlSecs;
+  constructor(file?: string, ttlSecs?: number) {
+    this.file = file || VersionCheckTTL.timestampFilePath();
+    try {
+      mkdirsSync(path.dirname(this.file));
+      fs.accessSync(path.dirname(this.file), fs.constants.W_OK);
+    } catch {
+      throw new Error(`Directory (${path.dirname(this.file)}) is not writable.`);
```

Comment:
Is this worth throwing? Shouldn't we flip a boolean that says "can't do any work, soz" ?

Reply:
The "can't do any work" message is printed by the `catch` block at the calling point of this constructor.

I did consider flipping a boolean and keeping it as a class member. This would mean the other methods on the class (like `hasExpired()`) return an additional state to the caller, which the caller will need to take care of.

Throwing an exception here seemed simpler and achieves the same experience.